tests/spellchecking/spellcheck.js

```diff
@@ -0,0 +1,275 @@
+/* bender-tags: editor,unit */
+/* bender-ckeditor-plugins: nanospell */
+
+'use strict';
+
+(function () {
+	bender.editor = {
+		config: {
+			enterMode: CKEDITOR.ENTER_P
+		}
+	};
+
+  bender.test({
+    assertHtml: function( expected, actual, msg ) {
+			assert.areEqual( bender.tools.fixHtml( expected ), bender.tools.fixHtml( actual ), msg );
+		},
+		setUp: function () {
+			this.server = sinon.fakeServer.create();
+			this.server.respondImmediately = true;
+
+			// for these tests we don't really care that much about the
+			// mock data, just that it returns something vaguely resembling the server call
+			var suggestions = {
+				"result": {
+					"appkes": ["apples"],
+					"pearrs": ["pears"],
+					"bannanas": ["bananas"],
+          "missspelling": ["misspelling"]
+				}
+			};
+
+			this.server.respondWith(
+				'/spellcheck/nano/',
+				JSON.stringify(suggestions)
+			);
+		},
+		tearDown: function () {
+			this.server.restore();
+			// reset the plugin and clear all spellcaches
+			this.editorBot.editor.execCommand('nanospellReset');
+		},
+    'test it can spellcheck a simple paragraph': function() {
+			var bot = this.editorBot,
+				tc = this,
+				editor = bot.editor,
+				resumeAfter = bender.tools.resumeAfter,
+				starterHtml = '<p>Paragraph with missspelling</p>';
+
+			bot.setHtmlWithSelection(starterHtml);
+
+			resumeAfter(editor, 'spellCheckComplete', function() {
+				var paragraph = editor.editable().findOne('p');
+
+				tc.assertHtml('<p>Paragraph with <span class="nanospell-typo">missspelling</span></p>', paragraph.getOuterHtml());
```

Comment:
This doesn't have to be done now,

But I think we could have used a utility function here:

```
getMarkedWords: function (editor) {
     ... return [contents of all span.nanospell-typo]
}
```

Then use the array matching assertion.   Unless we care specifically about the details of the html, which we might.  I can't think of a reason, though.